zebrad/src/components/sync/downloads.rs

```diff
@@ -17,11 +18,18 @@ use tokio::{sync::oneshot, task::JoinHandle};
 use tower::{hedge, Service, ServiceExt};
 use tracing_futures::Instrument;
 
-use zebra_chain::block::{self, Block};
+use zebra_chain::{
+    block::{self, Block},
+    chain_tip::ChainTip,
+};
 use zebra_network as zn;
+use zebra_state as zs;
 
 type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
 
+/// We allow an extra two checkpoints' worth of blocks in the verifier and state queues.
+const EXTRA_DOWNLOADS_LOOKAHEAD: usize = 2 * zebra_consensus::MAX_CHECKPOINT_HEIGHT_GAP;
```

Comment:
TODO: Increase this limit.

If some downloads, validation or state is slow, it's possible for Zebra to commit 2500 blocks (lookahead + 1 ExtendTips), then move on to downloading more blocks. (Sometimes the checkpoint verifier will commit 2000-2500 blocks at the same time.)

So I'm not sure how to manage the tradeoff between:
- making dropping blocks rare
- limiting the number of blocks in the queues
- resetting fast enough to re-use those blocks

We could reset sync if the height is within a small multiple of the limit, and ignore the block otherwise?